### DISCUSSION TITLE
why do I need to reinstall kbatch each time I start up my server?

### DISCUSSION THREAD

**Discussion by erthward** [microsoft/PlanetaryComputer#253] [Q&A] [answered]
The title is the essentially the question. 

Details: I log into my server, `cd` to my own directory of work, run `pip install kbatch` and follow the instructions [here](https://kbatch.readthedocs.io/en/latest/#install), and things seem to run fine. Then I leave a batch job running, log out, log in later to check on it, but I find that I have to run `pip install kbatch` again before I can use kbatch to display running jobs, logs, etc.

Am I doing something wrong?

Thanks!

**Comment by TomAugspurger** [comment, marked as answer]
`kbatch` isn't included in the [Hub's (Python) environment](https://planetarycomputer.microsoft.com/docs/overview/environment/#understanding-the-software-environment). We could add it, I suppose.

That said, I view `kbatch` as an alternative / complement to the interactive computing experience provided by the Hub. I would typically expect kbatch jobs to be submitted from outside of the Hub. Is there a strong reason to start a notebook server on the Hub just to submit a job?

**Reply by erthward** [reply]
Thanks! That makes so much sense. Hahaha, clearly I still haven't fully adapted to the cloud! I wasn't starting a notebook, of course, but I was still logging into the Hub just to run things from the terminal there. (I suppose the statement "The Hub includes a kbatch service" in the MSPC docs sort of seeded that idea. For dummies like me, it might be worth adding something to the effect, "One kbatch is installed on your own machine, you can then run...")
